src/OrchardCore/OrchardCore.Navigation.Core/ServiceCollectionExtensions.cs

```diff
@@ -16,4 +16,14 @@ public static IServiceCollection AddNavigation(this IServiceCollection services)
 
         return services;
     }
+
+    /// <summary>
+    /// Registers a navigation provider.
+    /// </summary>
+    /// <typeparam name="TProvider"></typeparam>
+    public static IServiceCollection AddNavigationProvider<TProvider>(this IServiceCollection services)
```

Comment:
I like the extension name, but while it's a single line it might not be worth creating it otherwise we can extension for everything in startup

Reply:
Good point. It just simplify the common provider registration like .AddDataMigration<> and extensions.

Reply:
I know what you mean bro, anyhow I never mind to have it

Reply:
> I never mind to have it

@hishamco then please resolve the comment and approve the PR.

Reply:
Let me wait for @Piedone or @sebastienros they might agree or disagree about adding an extension with a single line. If we don't hear any feedback until tomorrow we could merge

Reply:
While it doesn't reduce the volume of code (much), such extensions do help with reducing what you need to learn to utilize the OC API: You don't need to think about/look up examples for whether `INavigationProvider` should be scoped or singletons, for example.